qiskit_nature_pyscf/pyscf_ground_state_solver.py

```diff
@@ -0,0 +1,201 @@
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PySCFGroundStateSolver(GroundStateSolver):
+    """A PySCF-based GroundStateSolver for Qiskit Nature.
+
+    This class provides a ``GroundStateSolver`` for Qiskit Nature, leveraging the ``fci`` module of
+    PySCF. This is utility does not enable any Quantum algorithms to be used (since it replaces them
+    in the Qiskit workflow) but instead provides a useful utility for debugging classical
+    computational workflows based on Qiskit Nature.
+    More importantly, it provides a more efficient implementation of what Qiskit otherwise achieves
+    using a ``NumPyMinimumEigensolver`` in combination with a ``filter_criterion``. For non-singlet
+    spin ground states the setup using Qiskit components is a lot more involved, whereas this class
+    provides an easy-to-use alternative.
+
+    Here is an example use cas:
+
+    .. code-block:: python
+
+       from pyscf import fci
+
+       from qiskit_nature.second_q.drivers import MethodType, PySCFDriver
+       from qiskit_nature.second_q.transformers import ActiveSpaceTransformer
+
+       from qiskit_nature_pyscf import PySCFGroundStateSolver
+
+       driver = PySCFDriver(
+           atom="O 0.0 0.0 0.0; O 0.0 0.0 1.5",
+           basis="sto3g",
+           spin=2,
+           method=MethodType.UHF,
+       )
+       problem = driver.run()
+
+       transformer = ActiveSpaceTransformer(4, 4)
+
+       problem = transformer.transform(problem)
+
+       solver = PySCFGroundStateSolver(fci.direct_uhf.FCI())
+
+       result = solver.solve(problem)
+       print(result)
+
+    For more details please to the documentation of [PySCF](https://pyscf.org/) and
+    [Qiskit Nature](https://qiskit.org/documentation/nature/).
+    """
+
+    def __init__(self, solver: fci.direct_spin1.FCISolver) -> None:
+        """
+        Args:
+            solver: a FCI solver provided by PySCF.
+        """
+        super().__init__(None)
+        self._solver = solver
+
+    def solve(
+        self,
+        problem: BaseProblem,
+        aux_operators: dict[str, SparseLabelOp | SparsePauliOp] | None = None,
+    ) -> ElectronicStructureResult:
+        """Finds the ground-state of the provided problem.
+
+        This method is written to match the ``GroundStateSolver`` API of Qiskit Nature but it does
+        not support the evaluation of ``aux_operators`` and will ignore them.
+        It is also only capable of solving an ``ElectronicStructureProblem`` and will raise an error
+        if another problem object is provided.
+
+        Args:
+            problem: the problem instance to be solved.
+            aux_operators: **ignored**.
+
+        Raises:
+            RuntimeError: if a problem other than an ``ElectronicStructureProblem`` is provided.
+
+        Returns:
+            The interpreted result object in Qiskit Nature format.
+        """
+        if aux_operators is not None:
+            LOGGER.warning("Ignored.")
```

Comment:
```suggestion
            LOGGER.warning("aux_operators were ignored.")
```
Maybe a bit more description so if you were looking at the log it would be more evident than just something that sats Ignored and you need to refer back to the source.